Issue 10

davidritter opened the pull request:
This change addresses the issue described in:
https://github.com/welkineins/vscode-astyle/issues/10

davidritter committed:
ISSUE-10 Fix call to execFile with a file larger than 200KB will fail

Update the execFile call to specify the maxBuffer size to be tuneable
and larger than 200KB.  By default the plugin will now look up the
current file size and allocate 2x the buffer to allow room for
whitespace and formatting expansion.

This also fixes the issue where the plugin could not format files
that were large than 200KB.

Add a new configuation option to allow a user to tune the buffer
multiplier to custom values.

Update documentation.

davidritter committed:
ISSUE-10 Fix a comment

welkineins commented on extension.js:
```diff
@@ -25,16 +32,19 @@ class AstyleFormatter {
             astyleBinPath = astyleBinPath.replace(/\${workspaceRoot}/g, vscode.workspace.rootPath);
+            
+            // Assume that the formatting/expansion of the document could double it's size
+            const maxBufferSize = this.getFilesizeInBytes(document.fileName) * astyleMaxBufferMultiplier;
```
Hi @davidritter,

Thank you very much for fixing this issue!!
After review, I think it would be better to set max buffer size at least larger than default value `200*1024`.
I'll merge this PR and release new version soon later.

Thanks.

davidritter committed:
Accidentally inserted tabs.  Switch to spaces.

welkineins reviewed (changes requested):
Hi David,

I did forgot to send the comment.

I think the max buffer size should at least larger than default value `200*1024`, this can prevent failed to output error message for a very small file.

Please help to change this.

Thank you :)

davidritter committed:
Capture the default maxBuffer size for execFile. Choose the maximum of
calculated new multiplier buffer size for a larger file or use the default
value for execFile to avoid any pointless buffer resizes.
